src/main/scala/cognite/spark/v1/FlexibleDataModelCorePropertySyncRelation.scala

```diff
@@ -0,0 +1,91 @@
+package cognite.spark.v1
+
+import cats.effect.{Async, IO}
+import cognite.spark.v1.FlexibleDataModelBaseRelation.ProjectedFlexibleDataModelInstance
+import cognite.spark.v1.FlexibleDataModelRelationFactory.{
+  ViewCorePropertyConfig,
+  ViewSyncCorePropertyConfig
+}
+import com.cognite.sdk.scala.common.ItemsWithCursor
+import com.cognite.sdk.scala.v1.GenericClient
+import com.cognite.sdk.scala.v1.fdm.common.filters.FilterDefinition.HasData
+import com.cognite.sdk.scala.v1.fdm.instances._
+import fs2.Stream
+import org.apache.spark.sql.SQLContext
+import org.apache.spark.sql.sources._
+
+/**
+  * Flexible Data Model Relation for syncing Nodes or Edges with properties
+  *
+  * @param config common relation configs
+  * @param corePropConfig view core property config
+  * @param sqlContext sql context
+  */
+private[spark] class FlexibleDataModelCorePropertySyncRelation(
+    config: RelationConfig,
+    corePropConfig: ViewSyncCorePropertyConfig)(sqlContext: SQLContext)
+    extends FlexibleDataModelCorePropertyRelation(
+      config,
+      ViewCorePropertyConfig(
+        corePropConfig.intendedUsage,
+        corePropConfig.viewReference,
+        corePropConfig.instanceSpace)
+    )(sqlContext) {
+
+  override def getStreams(filters: Array[Filter], selectedColumns: Array[String])(
+      client: GenericClient[IO]): Seq[Stream[IO, ProjectedFlexibleDataModelInstance]] = {
+    val selectedInstanceProps = if (selectedColumns.isEmpty) {
+      schema.fieldNames
+    } else {
+      selectedColumns
+    }
+
+    val syncRequests = compatibleInstanceTypes(intendedUsage).map { instanceType =>
+      val hasData: Option[HasData] = viewReference.map { viewRef =>
+        HasData(List(viewRef))
+      }
+      val tableExpression = instanceType match {
+        case InstanceType.Edge => TableExpression(edges = Some(EdgeTableExpression(filter = hasData)))
+        case InstanceType.Node => TableExpression(nodes = Some(NodesTableExpression(filter = hasData)))
+      }
+      val sourceReference: Seq[SourceSelector] = viewReference
+        .map(
+          r =>
+            SourceSelector(
+              source = r,
+              properties = selectedInstanceProps.toIndexedSeq.filter(p =>
+                p != "startNode" && p != "endNode" && p != "space" && p != "externalId" && p != "type")))
+        .toSeq
+
+      InstanceSyncRequest(
+        `with` = Map("sync" -> tableExpression),
+        cursors = None,
+        select = Map("sync" -> SelectExpression(sources = sourceReference)))
+    }
+    syncRequests.distinct.map { sr =>
+      syncOut(sr).map(toProjectedInstance(_, selectedInstanceProps))
+    }
+  }
+
+  private def syncOut(syncRequest: InstanceSyncRequest)(
+      implicit F: Async[IO]): Stream[IO, InstanceDefinition] =
```

Comment:
just a thought, maybe scala sdk could be extended to produce stream and last cursor as it might be a useful utility, also iirc there's already some paging->stream support there